### ISSUE TITLE
Hope support in the style tag in HTML and .vue file

### ISSUE THREAD

**Issue by tjx666** [dzhavat/css-initial-value#3]
as title...

**Comment by dzhavat** [comment]
Hey, can you provide an example for how CSS looks like in a .vue file? I haven't tried Vue before.

**Comment by tjx666** [comment]
> Hey, can you provide an example for how CSS looks like in a .vue file? I haven't tried Vue before.

You can see my vue code in [codeSandbox example]( https://codesandbox.io/s/vue-template-2sqrt).

The effect in codeSandbox is the same as my local VSCode:
<img width="1439" alt="Snipaste_2019-08-18_18-37-47" src="https://user-images.githubusercontent.com/41773861/63223420-943ad300-c1e7-11e9-8b7e-1385ce0cd49f.png">

**Comment by dwelle** [comment]
If this ends up being implemented, we should also support [`.svelte`](https://svelte.dev) which uses almost identical css definitions:

```html
<style>
	p {
		color: purple;
		font-family: 'Comic Sans MS';
		font-size: 2em;
	}
</style>

<p>These styles...</p>
```

Both `.vue` and `.svelte` are effectively just `html` files with custom file extensions.

**Comment by dzhavat** [comment]
Support for both Vue and Svelte will be quite awesome. One problem with the extension right now is that it doesn't take the context (document) into consideration because it assumes that everything is CSS. Right now it just checks if a word matches a certain regex pattern.

`.vue` and `.svelte`, or other similar frameworks, mix CSS, HTML and JS so a property named `text: this is a test` will also match the regex pattern even though it won't return anything on hover.

So in order to make this work, there should be some logic that checks whether the hover is inside `<style>` or `style=""` and return the relevant information.

If you have any suggestions for how to do that, you're welcome to share them. I'll think about it as well.

**Comment by tjx666** [comment]
any progress?

**Comment by dzhavat** [comment]
Hey @tjx666 
I'm sorry but I haven't looked into this. Do you still need it?

**Comment by tjx666** [comment]
@dzhavat Yes, it would be helpful for vue and svelte users. Maybe we should also update the latest data.
@johnsoncodehk any simple way to support it in vue?

**Comment by johnsoncodehk** [comment]
I guess you can implement a plugin to https://github.com/volarjs/plugins.
